Text editor

rob-sullivan opened the pull request:
Added place image with url and highlight.js for code syntax highlighting. There is a style overriding the code syntax style. "Scoped" or !important in css didn't solve. For image I left out image upload as this works with storage buckets and I didn't want users to blow firebase limits. I also left out mentions and table features as this wasn't important for now. I didn't have access to javascript files storing api credentials so had to comment them out. I also didn't have a username login to test texteditor so had to adjust router.js to view it. Rob

eltonlin1998 commented:
I think scoped just prevents your text editor's style from leaking, but doesn't prevent other places' css from leaking into text editor. 

Do you know if the CSS leakage is from Vuetify? Any potential fixes?

eltonlin1998 commented:
By the way, you can create an account with any email and password, which would help with keeping router.js intact

rob-sullivan commented:
- set highlight.js to version 9.16.2 in package.json
- Following code syntax accepted: javascript, css, xml, python, java
- link to more in commented section above import. Use as required
- Signed up and removed debug in router

eltonlin1998 commented:
Reviewing - looks good so far. Meanwhile, to get around the API keys for Twilio and Algolia, create 2 files with explicitly fake API keys (I updated the contributing document with instructions): 
![image](https://user-images.githubusercontent.com/26662928/89720478-3a897500-da05-11ea-8408-37d0dcf7c185.png)

eltonlin1998 commented:
In terms of styling, there's an issue that the text editor no longer fills up all the available width.
![image](https://user-images.githubusercontent.com/26662928/89720545-bc799e00-da05-11ea-9aee-eabb5eac0488.png)

rob-sullivan commented:
This should be due to max-width under editor class in style. I'll correct now.

rob-sullivan commented:
@eltonlin1998 for your review.

eltonlin1998 commented:
One last thing: I'm having trouble figuring out how to use the code syntax highlighting. Is there an example I could follow somewhere?

rob-sullivan commented:
should be automatic. Type the code you want then left mouse click hold and drag to select it. When selected hit code. The text editor should detect what code you typed and give it syntax. More info here: [https://highlightjs.org/usage/](https://highlightjs.org/usage/)

eltonlin1998 commented:
I see, I kept using <> instead of the [] button. Would it be possible to have button texts / or simplify buttons down to the essentials  (e.g. on this Github text editor the features are limited). 

Now, while the highlighting works, after submission, it loses it highlighting: 
![image](https://user-images.githubusercontent.com/26662928/89721214-8dffc100-da0d-11ea-9f63-84ff6c5fcf1e.png)

rob-sullivan commented:
highlight.js is only implemented in text editor. I will have to go through components that display result from text editor and make sure it's implemented here as well (For example ExplanationDisplay.vue). I'll do this as a separate pull request.

eltonlin1998 commented:
I think it'd be better to make a Pull Request "complete", and have the changes in ExplanationDisplay.vue also in here. What do you think?

rob-sullivan commented:
agreed. mark as complete and I will push changes to ExplanationDisplay.vue here.

rob-sullivan commented:
Let me know when you're free to chat.

rob-sullivan commented:
**Highlight Code syntax working now on both create and display.**

1. Added editable prop to TextEditor component
2. Added editable() to watch and set TextEditor to readonly based on editable.
3. Removed <p> tag outputting expl.html and replaced with read only version of TextEditor component

rob-sullivan commented:
Reverted back algoliaCreds & twilioCreds.

rob-sullivan commented:
Will close. let me know if you ever need a hand again. All the best rob
